File: app/src/main/java/com/example/softmusic/room/MusicDataBase.java
package com.example.softmusic.room;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.example.softmusic.musicSong.MusicSong;
import com.example.softmusic.songList.MusicSongList;

@Database(entities = {MusicSongList.class, MusicSong.class,PlaylistSongCrossRef.class},version = 1,exportSchema = false)
public abstract class MusicDataBase extends RoomDatabase {
    private static MusicDataBase musicDataBase;

    public MusicDataBase() {

    }
    synchronized public static MusicDataBase getInstance(Context context) {
        if (musicDataBase == null){
            musicDataBase = Room.databaseBuilder(context,MusicDataBase.class,"music").build();
        }
        return musicDataBase;
    }

    public abstract MusicDao getMusicDao();
}

File: app/src/main/java/com/example/softmusic/room/MusicDataBase.kt
package com.example.softmusic.room

import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.example.softmusic.BaseApplication
import com.example.softmusic.entity.MusicSong
import com.example.softmusic.entity.MusicSongList
import com.example.softmusic.entity.PlaylistSongCrossRef

@Database(
    entities = [MusicSongList::class, MusicSong::class, PlaylistSongCrossRef::class],
    version = 5,
    exportSchema = false
)
abstract class MusicDataBase : RoomDatabase() {
    abstract val musicDao: MusicDao

    companion object {
        private val musicDataBase =
            Room.databaseBuilder(BaseApplication.context, MusicDataBase::class.java, "name")
                .fallbackToDestructiveMigration()
                .build()

        @Synchronized
        fun getInstance(): MusicDataBase {
            return musicDataBase
        }
    }
}